### ISSUE TITLE
Align values.yaml With Cortex Architecture Diagram

### ISSUE THREAD

**Issue by SudoBrendan** [cortexproject/cortex-helm-chart#121]
Greetings,

Thanks for creating this chart! It helped our team go from knowing Cortex was available, to standing it up in about a day's work, but I think aligning this chart with other official Cortex docs would have got us up faster - I'm curious on your ideas to make it better.

Naming conventions in `values.yaml` don't correlate 1:1 with [the official Cortex architecture diagram](https://cortexmetrics.io/docs/architecture/) and make it hard to parse what is optional for an MVP Cortex stack, and what is required. There's also similar names for top-level keys (`config` vs `configs`). Based on that diagram, I expect `values.yaml` to have the following required top-level keys:

- `config_file` (renamed from `config`, with a link in the values file to [this page](https://cortexmetrics.io/docs/configuration/configuration-file/), and a blurb about how anything specified here is rawly interpreted)
- `nginx`
- `distributor`
- `ingester`
- `query_frontend`
- `querier`
- `table_manager`

Additionally, the following keys should exist, with a sub-key of `enabled` and a default of `false`:

- `configs_api` (renamed from `configs`)
- `ruler`
- `alertmanager`

What do you think?

**Comment by drcrees** [comment]
Ran into this as well, was expecting to be able to disable "optional" services, e.g. `configs`, `ruler`, `alertmanager`

**Comment by hafe** [comment]
I think the POD names should reflect the architecture and not the image used by the POD. For example:
nginx => gateway
memcached-blocks-metadata => cache-blocks-metadata or blocks-metadata-cache or metadata-cache

replace all "memcached" by just cache

Not sure how such changes can be made with backward compatibility but the chart version is only 0.3
